Kafka/legos/kafka_get_topic_health/kafka_get_topic_health.py

```diff
@@ -37,40 +48,47 @@ def kafka_get_topic_health(handle, group_id: str, topics: list) -> Dict:
     :rtype: Dictionary containing the health status and number of messages by topic and partition
     """
 
-    consumer = KafkaConsumer(bootstrap_servers=handle.config['bootstrap_servers'], group_id=group_id)
-
-    topic_info = {}
-
-    # Get all available topics
-    available_topics = consumer.topics()
-
-    for tp in topics:
-        # Check if the topic exists
-        if tp in available_topics:
-            for partition in consumer.partitions_for_topic(tp) or []:
-                tp_obj = TopicPartition(tp, partition)
-                # Fetch the earliest and latest available offsets for the partition
-                earliest_offset = consumer.beginning_offsets([tp_obj])[tp_obj]
-                latest_offset = consumer.end_offsets([tp_obj])[tp_obj]
-
-                # Calculate the number of messages
-                number_of_messages = latest_offset - earliest_offset
-
-                # Store the number of messages and topic existence status in the result dictionary
-                topic_info.setdefault(tp, {})[partition] = {
-                    "number_of_messages": number_of_messages,
-                    "topic_exists": "Yes"
-                }
-        else:
-            # If topic doesn't exist, add a special entry to indicate this
-            topic_info[tp] = {
-                -1: {
-                    "number_of_messages": 0,
-                    "topic_exists": "No"
-                }
-            }
-    consumer.close()
-
-    return topic_info
+    admin_client = KafkaAdminClient(bootstrap_servers=handle.config['bootstrap_servers'])
+    topic_health_info = {}
+
+    # Determine which consumer groups to process
+    if group_id:
+        consumer_groups = [group_id]
+    else:
+        consumer_groups_info = admin_client.list_consumer_groups()
+        consumer_groups = [group[0] for group in consumer_groups_info]
+
+    # Prepare a cache for Kafka info to minimize calls
+    cached_kafka_info = {}
+
+    for group in consumer_groups:
+        consumer = KafkaConsumer(bootstrap_servers=handle.config['bootstrap_servers'], group_id=group)
+        group_topics = topics if topics else list(consumer.topics())
```

Comment:
this does not make sense to me. we should have only group as input parameters IMO